contracts/andromeda_gumball/src/contract.rs

```diff
@@ -0,0 +1,881 @@
+    let state = STATE.may_load(deps.storage)?;
+    require(state.is_some(), ContractError::PriceNotSet {})?;
+    // Automatically switch to opposite status
+    if status {
+        status = false;
+    } else {
+        status = true;
+    }
+    STATUS.save(deps.storage, &status)?;
+    Ok(Response::new().add_attribute("action", "Switched Status"))
+}
+fn execute_sale_details(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    price: Coin,
+    max_amount_per_wallet: Option<Uint128>,
+    recipient: Recipient,
+) -> Result<Response, ContractError> {
+    let contract = ADOContract::default();
+    let status = STATUS.load(deps.storage)?;
+    // Check status, can't change sale details while buying is allowed
+    require(!status, ContractError::Refilling {})?;
+    // Check authority
+    require(
+        contract.is_contract_owner(deps.storage, info.sender.as_str())?,
+        ContractError::Unauthorized {},
+    )?;
+    // Check valid amount
+    require(
+        price.amount > Uint128::from(0_u64),
+        ContractError::InvalidZeroAmount {},
+    )?;
+    // Check valid denomination
+    require(
+        price.denom == *"uusd",
+        ContractError::InvalidFunds {
+            msg: "Only uusd is allowed".to_string(),
+        },
+    )?;
+    // Check valid max amount per wallet
+    let max_amount_per_wallet = max_amount_per_wallet.unwrap_or_else(|| Uint128::from(1u128));
+
+    require(
+        max_amount_per_wallet > Uint128::from(0_u64),
+        ContractError::InvalidZeroAmount {},
+    )?;
+    // This is to prevent cloning price.
+    let price_str = price.to_string();
+
+    // Set the state
+    let state = State {
+        price,
+        max_amount_per_wallet,
+        recipient: recipient.clone(),
+    };
+    STATE.save(deps.storage, &state)?;
+    Ok(Response::new().add_attributes(vec![
+        attr("action", "switch status"),
+        attr("price", price_str),
+        attr("max_amount_per_wallet", max_amount_per_wallet),
+        attr(
+            "recipient",
+            recipient.get_addr(
+                deps.api,
+                &deps.querier,
+                contract.get_mission_contract(deps.storage)?,
+            )?,
+        ),
+    ]))
+}
+fn execute_mint(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    mint_msg: Box<MintMsg<TokenExtension>>,
+) -> Result<Response, ContractError> {
+    let status = STATUS.load(deps.storage)?;
+    // Can only mint when in "refill" mode, and that's when status is set to false.
+    require(!status, ContractError::NotInRefillMode {})?;
+    let contract = ADOContract::default();
+    // check authority
+    require(
+        contract.is_contract_owner(deps.storage, info.sender.as_str())?,
+        ContractError::Unauthorized {},
+    )?;
+    let mut list = LIST.load(deps.storage)?;
+
+    let config = CW721_CONTRACT.load(deps.storage)?;
+    // Add to list of NFTs
+    list.push(mint_msg.clone().token_id);
```

Comment:
Instead of cloning `mint_msg`, clone the `token_id`. So change to `mint_msg.token_id.clone()`.